README.md

```diff
@@ -14,54 +14,164 @@
-* **Lazy attributes**: some sensors will be passive, and will wait for the IoT Agent to request for data. For those measures, the IoT Agent will register itself in the Context Broker as a Context Provider (for all the lazy measures of that device), so if any component asks the Context Broker for the value of that sensor, its request will be redirected to the IoT Agent (that behaves as a NGSI10 Context Provider). 
+* **Active attributes**: are measures that are pushed from the device to the IoT agent. This measure changes will be
+sent to the Context Broker as updateContext requests over the device entity. NGSI queries to the context broker will
+be resolved in the Broker database.
 
-The following sequence diagram shows the different NGSI interactions an IoT Agent makes with the Context Broker, explained in the following subsections (using the example of a OMA Lightweight M2M device).
+* **Lazy attributes**: some sensors will be passive, and will wait for the IoT Agent to request for data. For those
+measures, the IoT Agent will register itself in the Context Broker as a Context Provider (for all the lazy measures of
+that device), so if any component asks the Context Broker for the value of that sensor, its request will be redirected
+to the IoT Agent (that behaves as a NGSI10 Context Provider). This operation will be synchronous from the customer
+perspective: the Context Broker won't return a response until de device has returned its response to the IoT Agent.
+
+* **Commands**: in this case, the interaction will begin by setting an attribute in the device's entity, for which the
+IoT Agent will be regitered as CP. The IoT Agent will return an immediate response to the Context Broker, and will be
+held responsible of contacting the device to perform the command itself, updating special `status` and `info` attributes
+in the entity as soon as it has any information of the command progress.
+
+The following sequence diagram shows the different NGSI interactions an IoT Agent makes with the Context Broker,
+explained in the following subsections (using the example of a OMA Lightweight M2M device).
 
 ![General ](https://raw.github.com/dmoranj/iotagent-node-lib/develop/img/ngsiInteractions.png "NGSI Interactions")
 
-Be aware that the IoT Agents are only required to support NGSI10 operations `updateContext` and `queryContext` in their standard formats (both in XML and JSON) but will not answer to NGSI9 operations (or NGSI convenience operations of any kind).
+Be aware that the IoT Agents are only required to support NGSI10 operations `updateContext` and `queryContext` in their
+standard formats (currently in JSON format; XML deprecated) but will not answer to NGSI9 operations
+(or NGSI convenience operations of any kind).
```

Comment:
This requirement only apply to IoTAgents, it does not apply to the way CBroker info is requested because CBroker will convert the query/update to the proper NGSI IoTagent needs, isn't it?